lib/baby_squeel/active_record/version_helper.rb

```diff
@@ -10,6 +10,11 @@ class VersionHelper
       #   ::ActiveRecord::VERSION::MAJOR > 7 ||
       #     ::ActiveRecord::VERSION::MAJOR == 7 && ::ActiveRecord::VERSION::MINOR >= 1
       # end
+
+      def self.at_least_7_2?
+        ::ActiveRecord::VERSION::MAJOR == 7 && ::ActiveRecord::VERSION::MINOR >= 2
```

Comment:
You need to add `::ActiveRecord::VERSION::MAJOR > 7 ||` if the changes also apply for Rails 8.0

Reply:
unclear if this will work on rails 8 at this time

Reply:
The name of the method suggests that it will apply to Rails 8.0.

I believe that the `Associations::AliasTracker` is not reverting to a simple connection.

Even if there are issues in Version 8, it would be simpler to start with the Rails 7.2 changes rather than the Rails 7.1 changes.

ActiveRecord 8.0.0.rc2 is green with the changes you made. 👍